src/geometamaker/config.py

```diff
@@ -0,0 +1,60 @@
+import logging
+import os
+
+import platformdirs
+
+from . import models
+
+
+LOGGER = logging.getLogger(__name__)
+
+DEFAULT_CONFIG_PATH = os.path.join(
+    platformdirs.user_config_dir(), 'geometamaker_profile.yml')
+
+
+class Config(object):
+    """Encapsulates user-settings such as a metadata Profile."""
+
+    def __init__(self, config_path=DEFAULT_CONFIG_PATH):
+        """Load a Profile from a config file.
+
+        Use a default user profile if none given.
+
+        Args:
+            config_path (str): path to a local yaml file
+
+        """
+        self.config_path = config_path
+        self.profile = models.Profile()
+
+        try:
+            self.profile = models.Profile.load(self.config_path)
+        except FileNotFoundError as err:
+            LOGGER.debug(err)
+            pass
+        # an invalid profile should raise a TypeError
+        except TypeError as err:
+            LOGGER.warning(err)
+            LOGGER.warning(
+                f'{self.config_path} contains an inavlid profile. '
+                'It will be ignored. You may wish to delete() it.')
+
+    def __repr__(self):
+        """Represent config as a string."""
+        return (
+            f'{self.config_path}: \n'
+            f'{self.profile}')
```

Comment:
Although I don't think there are any strong python conventions around this, I've seen many repr calls include the name of the class as a form of self-documentation, so something like `Config(config={self.config_path}, profile={self.profile})`, but I'm curious what you think!  Would it be useful to include this sort of self-documentation here?

Reply:
Thanks, good idea!